environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java

```diff
@@ -768,7 +768,9 @@ public WeldContainer initialize() {
                 // weld-se uses CommonForkJoinPoolExecutorServices by default
                 .add(EXECUTOR_THREAD_POOL_TYPE.get(), COMMON.toString())
                 // weld-se uses relaxed construction by default
-                .add(ConfigurationKey.RELAXED_CONSTRUCTION.get(), true);
+                .add(ConfigurationKey.RELAXED_CONSTRUCTION.get(), true)
+                 // We do not expect any EE components in SE
```

Comment:
I am not sure this is good idea. People do all kinds of combinations and customizations and this *might* break their current apps. I would leave it as opt-in.
Or maybe...wasn't there an option to detect EE services somehow? So we could make assumptions on EE component presence if this option is unset?

Reply:
You're probably right. For Weld SE there should be at least a possibility to disable this "feature". My concern is that optional features are rarely used.

Reply:
Ok, in that case we should figure out how significant is this optimization. If it's worth it, we can have it as default and provide a way to turn it off. If it's minor, we should do it the other way around. I agree with your concern though.

Reply:
It's probably minor but I'd like to have it enabled by default.